caluma/caluma_workflow/serializers.py

```diff
@@ -229,56 +171,24 @@ class CaseSerializer(SendEventSerializerMixin, serializers.ModelSerializer):
     )
 
     def validate(self, data):
-        form = data.get("form")
-        workflow = data["workflow"]
-
-        if form:
-            if (
-                not workflow.allow_all_forms
-                and not workflow.allow_forms.filter(pk=form.pk).exists()
-            ):
-                raise exceptions.ValidationError(
-                    f"Workflow {workflow.pk} does not allow to start case with form {form.pk}"
-                )
+        try:
+            data = models.Case.objects._validate(data)
+        except Exception as e:
+            raise exceptions.ValidationError(str(e))
 
         return super().validate(data)
 
     @transaction.atomic
     def create(self, validated_data):
         user = self.context["request"].user
-        parent_work_item = validated_data.get("parent_work_item")
-        validated_data["status"] = models.Case.STATUS_RUNNING
-
-        form = validated_data.pop("form", None)
-        if form:
-            validated_data["document"] = Document.objects.create(
-                form=form, created_by_user=user.username, created_by_group=user.group
-            )
 
-        if parent_work_item:
-            case = parent_work_item.case
-            while hasattr(case, "parent_work_item"):
-                case = case.parent_work_item.case
-            validated_data["family"] = case
+        validated_data = models.Case.objects._pre_create(validated_data, user)
 
-        instance = super().create(validated_data)
-
-        # Django doesn't save reverse one-to-one relationships automatically:
-        # https://code.djangoproject.com/ticket/18638
-        if parent_work_item:
-            parent_work_item.child_case = instance
-            parent_work_item.save()
-
-        workflow = instance.workflow
-        tasks = workflow.start_tasks.all()
-
-        work_items = bulk_create_work_items(tasks, instance, user)
-
-        self.send_event(events.created_case, case=instance)
-        for work_item in work_items:  # pragma: no cover
-            self.send_event(events.created_work_item, work_item=work_item)
+        case = super().create(validated_data)
```

Comment:
Instead of doing the "pre_create", "create", "post_create" dance, we should call 
`case = models.Case.objects.start(...)` instead, cleaning up the parameters so the method can properly work with it.